File: week-04/day-2_Classes/src/Animal_Farm/Farm.java
package Animal_Farm;

        /*Create a Farm class
        it has list of Animals
                -it has slots which defines the number of free places for animals
                -breed() -> creates a new animal if there's place for it
                -slaughter() -> removes the least hungry animal*/

import java.util.ArrayList;

public class Farm {
  ArrayList<Animal> arrayListFarm;

  private int slots = 5;

  Farm(ArrayList<Animal> arrayListFarm) {
    this.arrayListFarm = arrayListFarm;
  }

  public void breed() {
    if (slots > 0) {
      arrayListFarm.add(new Animal());
      slots--;
    }
  }

  public void slaughter() {
    int index = 0;
    for (int i = 1; i < arrayListFarm.size(); i++) {
      int leastHungry = arrayListFarm.get(0).hunger;
      if (arrayListFarm.get(i).hunger < leastHungry) {
        index = i;
      }
    }
    arrayListFarm.remove(index);
  }

/*  public ArrayList<Animal> arrayListFarm() {
    return arrayListFarm();
  }*/

/*  public ArrayList<Animal> bread() {
    if (slots > 0) {
      //Animal breadAn = new Animal();
      slots--;
      arrayListFarm().add(new Animal());
    }
    return arrayListFarm();
  }*/
}

File: C#/classes/Farm.cs
using System.Collections.Generic;

class Farm
{
    public int Slots { get; set; }
    public List<Animal> animals;

    public Farm()
    {
        this.Slots = 2;
        this.animals = new List<Animal>();
    }

    public void Breed()
    {
        if (Slots > 0)
        {
            Animal toCreate = new Animal();
            animals.Add(toCreate);
            Slots--;
            System.Console.WriteLine("New animal was created!");
        }
        else
        {
            System.Console.WriteLine("There is no place for a new animal!");
        }

    }

    public void Slaughter()
    {
        int tempHunger = animals[0].Hunger;
        int index = 0;
        for (int i = 1; i < animals.Count; i++)
        {
            if (animals[i].Hunger < tempHunger)
            {
                tempHunger = animals[i].Hunger;
                index = i;
            }
        }
        System.Console.WriteLine("We have so many animals : " + animals.Count);
        System.Console.WriteLine("The animal at index {0} with hunger {1} will be removed.", index, tempHunger);
        animals.RemoveAt(index);
        System.Console.WriteLine("We have so many animals : " + animals.Count);
    }

    public void Add(Animal animal)
    {
        animals.Add(animal);
    }
}